sections/Stake/StakeBox/StakeBox.tsx

```diff
@@ -70,65 +94,87 @@ const ActionBox: FC<ActionBoxProps> = ({
 		}
 	}, [isApproved, onApprove, action, onSubmit, inputValue, symbol]);
 
-	useEffect(() => {
-		if (txState !== TransactionStatus.PRESUBMIT) setShowTxModal(false);
-	}, [txState]);
+	const isActionButtonDisabled: boolean = useMemo(() => {
+		if (!walletAddress || !inputValue) return true;
```

Comment:
lets add brackets around every if statement. It is generally safer to program that way although i know its generally not a big deal either way